src/js/claims-status/containers/AppealInfo.jsx

```diff
@@ -1,20 +1,69 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Link } from 'react-router';
+
+import moment from 'moment';
+
+import Breadcrumbs from '../components/Breadcrumbs';
 import AppealsV2TabNav from '../components/appeals-v2/AppealsV2TabNav';
 
-const AppealInfo = ({ params, children }) => {
-  const appealId = params.id;
-  return (
-    <div>
-      <AppealsV2TabNav appealId={appealId}/>
-      {children}
-    </div>
-  );
-};
+function isolateAppeal(state, id) {
+  const claimsState = state.disability.status;
+  const { appeals } = claimsState.claims;
+  const appeal = appeals.find(a => a.id === id);
+
+  // append starting event for post-remand and post-cavc remand appeals
+  // NOTE: This is business logic pulled from v1 that we don't fully understand yet.
+  //  Also, having this logic in mapStateToProps is less than ideal; we want to
+  //  move it out when we know where it should live. Maybe just a helper.
+  if (appeal && appeal.attributes.type !== 'original' && appeal.attributes.prior_decision_date) {
+    appeal.attributes.events.unshift({
+      type: appeal.attributes.type === 'post_cavc_remand' ? 'cavc_decision' : 'bva_remand',
+      date: appeal.attributes.prior_decision_date,
+    });
+  }
+
+  return appeal;
+}
+
+export class AppealInfo extends React.Component {
```

Comment:
Does this need to be a class-based component? I *think* functional components are `connect`able as well

Reply:
Um, I'm not sure. I had it in my head that they weren't, so I just made it a class... :man_shrugging:

I can try keeping it functional and :crossed_fingers: just for fun.

Reply:
Good news! You were right. Totally works. 💥